addons/snailmail/static/src/models/message.js

```diff
@@ -35,11 +35,14 @@ addRecordMethods('Message', {
      * Opens the action about 'snailmail.letter' missing fields.
      */
     async openMissingFieldsLetterAction() {
-        const letterIds = await this.async(() => this.messaging.rpc({
+        const letterIds = await this.messaging.rpc({
             model: 'snailmail.letter',
             method: 'search',
             args: [[['message_id', '=', this.id]]],
-        }));
+        });
+        if (!this.exists()) {
+            return;
+        }
```

Comment:
should probably still do action even if record (possibly view record later) gets deleted